docs/administration.building-blocks.md

```diff
@@ -0,0 +1,127 @@
+  in meshStack, but no underlying resources (e.g. in a cloud platform) will be deleted. You have to do this yourself.
+- Delete Resources: this option will also clean up the underlying resources. The workings depend on the implementation
+  type, but e.g. for Terraform meshStack will call `terraform destroy` on the configuration files.
+
+### Dependencies
+
+> Please note that this the use of dependencies is completely optional and this step can be skipped.
+
+In some cases you might find that you have a multiple building block definitions, and one depends on the other.
+For example, you might have a Virtual Network, which is a prerequisite for a Virtual Machine.
+
+You can model and enforce such dependencies in meshStack using the Building Blocks feature.
+
+In the wizard, on the "Dependencies" page, you have to select the building block(s) that the _about to be created_
+building block depends on. So in the example above you would have to do the following:
+
+1. Create the Virtual Network Building Block without any dependencies.
+2. Create the Virtual Machine Building Block and select the Virtual Network as a dependency.
+
+> For Terraform users: you can provide the output of the Virtual Network Building Block as input for the Virtual
+> Machine building block.
+
+### Building Block Inputs
+
+As with a lot of configuration, input is required. Building Blocks in meshStack can collect as much input as you want,
+in many ways possible.
+
+There are different kinds of inputs, with different behavior, which can be decided under the "Source" section. You can
+read in the meshPanel what each kind of input does and how it behaves.
+
+The inputs' behavior can be further decided:
+
+- Provide as: you can decide whether the input is regularly provided or as an environment variable.
+  (only applies to Terraform Building Blocks)
+- Sensitivity: you can decide that an input is sensitive (e.g. a secret). It will then be encrypted in meshStack and not
+  shown to any users.
+
+### Building Block Outputs (Terraform-only)
+
+You can also extract the Terraform outputs into meshStack. This is useful in two scenarios:
+
+1. You want to provide an output variable to a dependent building block. As the example that was mentioned above, if
+   you have a virtual machine that depends on the IP Address (? - check) of a virtual network, you can output the
+   variable and provide it as an input to the dependent virtual machine building block.
+2. You want the customer user to be able to see the output result. This might be useful if there is an output that is
+   important to the customer user. They will be able to read the value in their tenant control plane.
+
+Last but not least, save your building block definition, and it will be ready to be consumed by your customer users!
+
+## Developing & Deploying a Terraform Building Block
+
+For Terraform blocks, meshStack will run the required Terraform in the background.
+
+### How to debug
```

Comment:
d: I think at least with a next PR we should also mention the option to rerun a building block (e.g. if it failed and the operator now fixed the underlying issue). And we should mention that blocks can also be deleted